caddyconfig/httpcaddyfile/httptype.go

```diff
@@ -133,14 +140,19 @@ func (st ServerType) Setup(originalServerBlocks []caddyfile.ServerBlock,
 				return nil, warnings, fmt.Errorf("%s:%d: unrecognized directive: %s", tkn.File, tkn.Line, dir)
 			}
 
-			results, err := dirFunc(Helper{
+			h := Helper{
 				Dispenser:    caddyfile.NewDispenser(segment),
 				options:      options,
 				warnings:     &warnings,
 				matcherDefs:  matcherDefs,
 				parentBlock:  sb.block,
 				groupCounter: gc,
-			})
+				state:        make(map[string]interface{}),
+			}
+			h.state["tlsCertTags"] = tlsCertTags
+
+			results, err := dirFunc(h)
+
```

Comment:
I like to keep `err != nil` checks immediately after the line that can produce the error, for readability

Reply:
Sure thing